Source/WebCore/svg/SVGAnimationElement.cpp

```diff
@@ -428,10 +428,8 @@ float SVGAnimationElement::calculatePercentFromKeyPoints(float percent) const
 {
     const auto& keyTimes = m_keyTimesFromAttribute;
 
-    ASSERT(!m_keyPoints.isEmpty());
-    ASSERT(calcMode() != CalcMode::Paced || animationMode() == AnimationMode::Path);
-    ASSERT(keyTimes.size() > 1);
-    ASSERT(m_keyPoints.size() == keyTimes.size());
+    if (m_keyPoints.isEmpty() || (calcMode() == CalcMode::Paced && animationMode() != AnimationMode::Path) || keyTimes.size() <= 1 || m_keyPoints.size() != keyTimes.size())
+        return percent;
```

Comment:
Do we understand why we end up having any of these checks to be true?

According to the existing assertions these checks can't be true. Your change will silence these assertions forever. Are we sure we are not hiding an existing bug somewhere else?

Reply:
Fair point. I was trying to harden the function but for this one specific crash the issue is that `m_keyPoints.size()` and `keyTimes.size()` are < 2, which leads to the OOB. From the [commit message](https://github.com/WebKit/WebKit/pull/27204) of the change that introduced this issue,  this assertion is no longer guaranteed, at least for certain modes.